include/zephyr/pm/device.h

```diff
@@ -628,13 +628,33 @@ bool pm_device_is_powered(const struct device *dev);
  * @ref PM_DEVICE_STATE_OFF, or @ref PM_DEVICE_STATE_SUSPENDED if device can
  * never be powered off.
  *
+ * @note If CONFIG_PM_DEVICE=n, PM_DEVICE_ACTION_TURN_ON followed by
+ * PM_DEVICE_ACTION_RESUME will be invoked unconditionally.
+ *
  * @param dev Device instance.
  * @param action_cb Device PM control callback function.
  * @retval 0 On success.
  * @retval -errno Error code from @a action_cb on failure.
  */
 int pm_device_driver_init(const struct device *dev, pm_device_action_cb_t action_cb);
 
+/**
+ * @brief Deinit a device driver
+ *
+ * @details Ensures device is either SUSPENDED or OFF, and will revert the actions of
+ * pm_device_driver_init() if this is the case. Must be called at the beginning of a
+ * driver's deinit function.
```

Comment:
It says must be called by driver, okay, but is it illegal to be called anywhere else ? if so, should that be clarified?